packages/macros/README.md

```diff
@@ -8,61 +8,179 @@ Traditionally, Ember addons have a lot of power to run arbitrary code during the
 
-   ```js
-   let app = new EmberApp(defaults, {
-     '@embroider/macros': {
-       // this is how you configure your own package
-       setOwnConfig: {
-         // your config goes here
-       },
-       // this is how you can optionally send configuration into your
-       // dependencies, if those dependencies choose to use
-       // @embroider/macros configs.
-       setConfig: {
-         'some-dependency': {
-           // config for some-dependency
-         },
-       },
-     },
-   });
-   ```
+// will compile to:
 
-## Setting Configuration: from an Ember Addon
+let doesFooExist = true; // or false if the dependency was not satisfied
+```
 
-1. Add `@embroider/macros` as `dependency`.
-2. In `index.js`, do:
+We can use this macro along with the `macroCondition` and `importSync` macro's from above to do something more complex:
+
+```js
+import { macroCondition, dependencySatisfies, importSync } from '@embroider/macros';
+
+if (macroCondition(dependencySatisfies('ember-qunit', '*'))) {
+  return importSync('ember-qunit');
+} else if (macroCondition(dependencySatisfies('ember-mocha', '*'))) {
+  return importSync('ember-mocha');
+}
+```
+
+```hbs
+{{macroDependencySatisfies 'qunit' '^2.8.0'}}
+```
+
+### getOwnConfig, getConfig, and getGlobalConfig
+
+A common pattern is to have a set of configuration properties that you define (or a consumer defines for you) which you base certain build time conditions around. This is achieved via the `getOwnConfig`, `getConfig`, and `getGlobalConfig` macros (depending on which config you want to read).
+
+```js
+module.exports = {
+  name: require('./package').name,
+  options: {
+    '@embroider/macros': {
+      setOwnConfig: {
+        themeColor: 'red',
+      },
+    },
+  },
+  included() {
+    this._super.included.apply(this, arguments);
+    this.options['@embroider/macros'].setOwnConfig.shouldIncludeMinifiedLibrary = false;
+  },
+};
+```
+
+```js
+import { getOwnConfig, importSync, macroCondition } from '@embroider/macros';
+
+if (macroCondition(getOwnConfig().shouldIncludeMinifiedLibrary)) {
+  importSync('minified-library');
+} else {
+  importSync('unminified-library');
+}
+```
+
+```hbs
+<button class="{{macroGetOwnConfig "themeColor"}}">My Themed Button</button>
+```
+
+## Real world examples
+
+Below are a list of addons that have started using `@embroider/macros` so that you can get a feel for common use cases that can be solved via the macro system.
 
-   ```js
-   module.exports = {
-     name: require('./package').name,
-     options: {
-       '@embroider/macros': {
-         setOwnConfig: {
-           // your config goes here
-         },
-         setConfig: {
-           'some-dependency': {
-             // config for some-dependency
-           },
-         },
-       },
-     },
-   };
-   ```
+- [ember-exam](https://github.com/ember-cli/ember-exam)
+- [ember-bootstrap](https://github.com/kaliber5/ember-bootstrap)
```

Comment:
If you want to add another example, I use macros in tests only for https://github.com/kaliber5/ember-stargate, basically as a replacement for `ember-compatibility-helpers`

Reply:
👍  will add